packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -30,20 +30,26 @@ const clientVersionHeaderKey = 'apollographql-client-version';
 export const plugin = <TContext>(
   options: EngineReportingOptions<TContext> = Object.create(null),
   addTrace: (args: AddTraceArgs) => Promise<void>,
-  // schemaHash: string,
 ): ApolloServerPlugin<TContext> => {
   const logger: Logger = options.logger || console;
   const generateClientInfo: GenerateClientInfo<TContext> =
     options.generateClientInfo || defaultGenerateClientInfo;
 
-
   return {
     requestDidStart({
       logger: requestLogger,
       schemaHash,
       metrics,
       request: { http, variables },
     }) {
+      // If the options are false don't do any metrics timing.
+      if (options.traceReporting === false) {
+        // If we are running in gateway mode don't send the ftv1 trace
+        metrics.captureTraces = false;
+        return {};
```

Comment:
Did you find it necessary to return an object literal rather than just `return` (i.e. `undefined`)? (If so, that would seem to be a bug that I should fix.)

```suggestion
        return;
```

Reply:
Looks like I have to return an object. We throw in `pluginTestHarness.ts:121` if there are no listeners returned.